frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -499,107 +614,32 @@ function ReaderHighlight:onShowHighlightDialog(page, index)
-                enabled = self.hold_pos ~= nil,
-            },
-        },
-        {
-            {
-                text = C_("Text", "Copy"),
-                enabled = Device:hasClipboard(),
-                callback = function()
-                    Device.input.setClipboardText(cleanupSelectedText(self.selected_text.text))
-                end,
-            },
-            {
-                text = _("View HTML"),
-                enabled = not self.ui.document.info.has_pages,
-                callback = function()
-                    self:viewSelectionHTML()
-                end,
-            },
-        },
-        {
-            {
-                text = _("Wikipedia"),
-                callback = function()
-                    UIManager:scheduleIn(0.1, function()
-                        self:lookupWikipedia()
-                        -- We don't call self:onClose(), we need the highlight
-                        -- to still be there, as we may Highlight it from the
-                        -- dict lookup widget
-                    end)
-                end,
-            },
-            {
-                text = _("Dictionary"),
-                callback = function()
-                    self:onHighlightDictLookup()
-                    -- We don't call self:onClose(), same reason as above
-                end,
-            },
-        },
-        {
-            {
-                text = _("Translate"),
-                callback = function()
-                    self:translate(self.selected_text)
-                    -- We don't call self:onClose(), so one can still see
-                    -- the highlighted text when moving the translated
-                    -- text window, and also if NetworkMgr:promptWifiOn()
-                    -- is needed, so the user can just tap again on this
-                    -- button and does not need to select the text again.
-                end,
-            },
-            {
-                text = _("Search"),
-                callback = function()
-                    self:onHighlightSearch()
-                    UIManager:close(self.highlight_dialog)
-                end,
-            },
-        },
-    }
-    if self.selected_link ~= nil then
-        table.insert(highlight_buttons, { -- for now, a single button in an added row
-            {
-                text = _("Follow Link"),
-                callback = function()
-                    local link = self.selected_link.link or self.selected_link
-                    self.ui.link:onGotoLink(link)
-                    self:onClose()
-                end,
-            },
-        })
+function ReaderHighlight:addToHighlightDialog(idx, fn_button)
+    -- fn_button is a function that takes the ReaderHighlight instance
+    -- as argument, and returns a table describing the button to be added.
+    self._highlight_buttons[idx] = fn_button
+    local n = 0
+    for _ in pairs(self._highlight_buttons) do
+        n = n + 1
```

Comment:
(It's short so I don't mind - but we have `util.tableSize(t)`).
I might be confused by the diff - but it seems you don't do anything with that local n ?!

Reply:
Sorry, it was for debugging and I forgot it.